pyiron_base/database/generic.py

```diff
@@ -372,6 +372,33 @@ def get_items_sql(self, where_condition=None, sql_statement=None):
             output_list += [dict(zip(col.keys(), tmp_values))]
         return output_list
 
+    def _truncate_chem_formula(self, chem_formula):
+        """
+        this functions truncate the chemical formula to the last element,
+        up to the limit of the database.
+        arg:
+        chem_form(str): the chemical formula
+        """
+        acceptable_len=30
```

Comment:
Can we pull that out and use the value from the database column definition or is that defined somewhere else?  Leave it here if it takes more than 5 minutes.

Reply:
Yeah, it's [here](https://github.com/pyiron/pyiron_base/blob/e59f4682ba6228f725c0a8d10c8b2ed93551890e/pyiron_base/database/generic.py#L116), but it's hard-coded inline with the table instantiation. It could be refactored out as a parameter and referenced though...

Reply:
I have added a variable to the class, `_chem_formula_lim_length`, which takes the value of 30 and defines the length of the chamical formula in the db.